arklex/env/tools/tools.py

```diff
@@ -118,8 +118,20 @@ def __init__(
         self.node_specific_data: dict[str, Any] = {}
         self.fixed_args = {}
         self.properties: dict[str, dict[str, Any]] = {}
-        # TODO: check with voicebot setup
-        # self.openai_slots: list[dict[str, Any]] = self._format_slots(slots)
+
+    def copy(self) -> "Tool":
+        """Create a copy of this tool instance.
+
+        Returns:
+            Tool: A new Tool instance with the same configuration but independent state.
+        """
+        new_tool = Tool(
+            func=self.func,
+            name=self.name,
+            description=self.description,
+            slots=self.slots,
+        )
+        return new_tool
```

Comment:
1. Avoid the extra variable
`return Tool(...`
2. Dont we have to copy fixed args, auth and other data?